src/registry/datetime/validation/types.ts

```diff
@@ -0,0 +1,13 @@
+import {DateTimeComponentSchema} from '@open-formulieren/types';
+import {DateConstraintConfiguration} from '@open-formulieren/types/lib/formio/dates';
+
+import {FilterByValueType} from '@/types';
+
+// A bunch of derived types from the DateTimeComponentSchema that makes working with the
+// schema a bit more readable while keeping everything exhaustive and type safe.
+type AllDateExtensions = Required<NonNullable<DateTimeComponentSchema['openForms']>>;
+
+export type AllModes = DateConstraintConfiguration['mode'];
+export type NonEmptyModes = Exclude<AllModes, ''>;
+export type DateConstraintKey = keyof FilterByValueType<AllDateExtensions, {mode: AllModes}>;
+export type AllPossibleConstraints = AllDateExtensions[DateConstraintKey];
```

Comment:
Should there also be a type for the `includeToday`?

Reply:
`datetime` doesn't support `includeToday` :)
`date` does!